commands/sketch/new.go

```diff
@@ -34,6 +35,10 @@ void loop() {
 }
 `)
 
+// sketchNameMaxLength could be part of the regex, but it's intentionally left out for clearer error reporting
+var sketchNameMaxLength = 63
+var sketchNameValidationRegex = regexp.MustCompile(`^[0-9a-zA-Z]{1}[0-9a-zA-Z_\.-]*$`)
```

Comment:
```suggestion
var sketchNameValidationRegex = regexp.MustCompile(`^[0-9a-zA-Z][0-9a-zA-Z_\.-]*$`)
```

The `{1}` is superfluous. It made sense for the sake of consistency in the Arduino IDE regex, which also enforces the maximum length, but this regex doesn't do that.

Since the regex is exposed to the user in the error message, I think having the less obvious and less common quantifier syntax in the regex makes the error message more difficult to understand for a user without a good knowledge of regular expressions.

Reply:
Should be better now, I also found out empty sketch names were not really handled before. Now they are

Reply:
@Bikappa the regex is still the same as it was when I did my review.

Reply:
Should be gone for real now